tests/kafkatest/services/security/security_config.py

```diff
@@ -258,8 +259,10 @@ def enable_sasl(self):
     def enable_ssl(self):
         self.has_ssl = True
 
-    def enable_security_protocol(self, security_protocol):
+    def enable_security_protocol(self, security_protocol, sasl_mechanism = None):
```

Comment:
Do we make use of this new argument anywhere? I can't seem to find any

Reply:
Good catch!  I missed leveraging this, and it results in `sasl.enabled.mechanisms=` (i.e. blank) on the first of the two rolls.  The test was still passing, but I think that's just luck.  I've fixed it in `kafka.py` -- we now pass in the SASL mechanism, and we are now seeing `sasl.enabled.mechanisms=PLAIN` as expected.